server/mongodb/actions/applications.js

```diff
@@ -64,6 +64,39 @@ async function deleteApplication(id) {
     });
 }
 
+async function updateApplicationState(id, state) {
+  await mongoDB();
+  let result = {};
+
+  try {
+    const curObject = await Application.findOne({ _id: id }, { status: 1 });
+    if (curObject.status < 3) {
+      result = await Application.findOneAndUpdate({ _id: id }, { status: state, decision: null },
+        { upsert: false });
+    } else {
+      result = await Application.findOneAndUpdate({ _id: id }, { status: state },
+        { upsert: false });
```

Comment:
Change to return updated application instead of old
```suggestion
        { upsert: false, new: true });
```